src/main/java/com/powsybl/openloadflow/sensi/AbstractSensitivityAnalysis.java

```diff
@@ -585,9 +586,14 @@ protected List<SensitivityFactorGroup<V, E>> createFactorGroups(List<LfSensitivi
         return new ArrayList<>(groupIndexedById.values());
     }
 
-    protected List<ParticipatingElement> getParticipatingElements(Collection<LfBus> buses, LoadFlowParameters.BalanceType balanceType, OpenLoadFlowParameters openLoadFlowParameters) {
+    protected List<ParticipatingElement> getParticipatingElements(LfNetwork network, LoadFlowParameters.BalanceType balanceType, OpenLoadFlowParameters openLoadFlowParameters) {
+        return getParticipatingElements(network, Collections.emptySet(), balanceType, openLoadFlowParameters);
+    }
+
+    protected List<ParticipatingElement> getParticipatingElements(LfNetwork network, Set<LfBus> disabledBuses, LoadFlowParameters.BalanceType balanceType, OpenLoadFlowParameters openLoadFlowParameters) {
         ActivePowerDistribution.Step step = ActivePowerDistribution.getStep(balanceType, openLoadFlowParameters.isLoadPowerFactorConstant());
-        List<ParticipatingElement> participatingElements = step.getParticipatingElements(buses);
+        List<LfBus> nonDisabledBuses = network.getBuses().stream().filter(Predicate.not(disabledBuses::contains)).collect(Collectors.toList());
```

Comment:
So to avoid building the bus set of main CC in sensitiity analysis  (see `ConnectivityAnalysisResult`), we need to build to nonDisabledBuses (which is tie consuming) each time we distribute the slack which is used everywhere in the code (simple LF, AS, sensi) and many, many times.
Are we sure this refactoring is a good idea?